### DISCUSSION TITLE
How to have multiple components loaded on the same route path based on data?

### DISCUSSION THREAD

**Discussion by okrymus** [nuxt/nuxt#16597] [Questions]
Are there any way to do a similar to this example with Nuxt 3?

I have app where user can login in different roles, eg. seller, buyer and admin. For each user I'd like to show dashboard page on the same path, eg. http://localhost:8080/dashboard However, each user will have different dashboard defined in different vue components, eg. SellerDashboard, BuyerDashboard and AdminDashboard.

So basically, when user opens http://localhost:8080/dashboard vue app should load different component based on the user role (which I store in vuex). Similarly, I'd like to have this for other routes. For example, when user goes to profile page http://localhost:8080/profile app should show different profile component depending on the logged in user.

So I'd like to have the same route for all users roles as opposed to have different route for each user role, eg. I don't want user role to be contained in url like following: http://localhost:8080/admin/profile and http://localhost:8080/seller/profile etc...

How can I implement this scenario with vue router?

```js
import Vue from "vue";
import VueRouter from "vue-router";
import Home from "../views/Home.vue";
import store from "../store";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: () => {
      switch (store.state.userRole) {
        case "admin":
          return import("../components/AdminDashboard");
        case "buyer":
          return import("../components/BuyerDashboard");
        case "seller":
          return import("../components/SellerDashboard");
        default:
          return Home;
      }
    }
  }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

export default router;
```

**Comment by TheAlexLichter** [comment]
Instead of implementing it on the router level, why not setting the components inside the page components (e.g. `pages/dashboard.vue`)?
You could also add a composable that makes differentiating between roles and reuse the logic throughout them.

**Reply by okrymus** [reply]
Thank you for your reply. Do you know how I can use definePageMeta inside the components inside the page components

**Reply by TheAlexLichter** [reply]
AFAIK that is not possible 🙁
